src/state/Maps/reducers.js

```diff
@@ -241,6 +242,53 @@ const removeMapLayers = (state, mapKey, layerKeys) => {
 	}
 };
 
+/**
+ * Remove layers satisfying filter from map
+ * @param state {Object}
+ * @param mapKey {string}
+ * @param filter {object}
+ * @return {Object} Updated state
+ */
+const removeMapLayersByFilter = (state, mapKey, filter) => {
+	if (!_isEmpty(filter) && mapKey) {
+		const mapLayers = state.maps[mapKey]?.data?.layers;
+		const mapLayersByFilterIndexes = [];
+		mapLayers?.forEach((l, i) => {
+			if (_isMatch(l, filter)) {
+				mapLayersByFilterIndexes.push(i);
+			}
+		});
+
+		if (mapLayersByFilterIndexes?.length) {
+			let updatedLayers = [...state.maps[mapKey]?.data.layers];
+			mapLayersByFilterIndexes.forEach((layerIndex, i) => {
+				updatedLayers = stateManagement.removeItemByIndex(
+					updatedLayers,
+					layerIndex - i
```

Comment:
The double traversing by both layerIndex and i seems like a bug breeding ground, in addition to being unclear.
Woud be better to remove at once. Or even better, build a copy in the first cycle (line 256) and use it if matches found?

Reply:
Thats right, I was counting, if you have 200 layers in map and you want to remove 5 layers by filter, then in current solutions should be 205 cycles, but your solution shoud pass "only" 200 cycles.

But you are right. minute.

Reply:
improved here: c8cb64a2a81c3d34c98a2ed4feaeabf08a0558b6